docs/source/java/vector_schema_root.rst

```diff
@@ -0,0 +1,60 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+================
+VectorSchemaRoot
+================
+A :class:`VectorSchemaRoot` is a container that can hold batches, batches flow through :class:`VectorSchemaRoot`
+as part of a pipeline. Note this is different from other implementations (i.e. in C++ and Python,
+a :class:`RecordBatch` is a collection of equal-length vector instances and was created each time for a new batch).
+
+The recommended usage for :class:`VectorSchemaRoot` is creating a single :class:`VectorSchemaRoot` based on
+ the known schema and populated data over and over into the same VectorSchemaRoot in a stream of batches rather than
+ create a new :class:`VectorSchemaRoot` instance each time (see ``Flight`` or ``ArrowFileWriter`` for better
```

Comment:
can you provide specific links to the code for flight?

How does one transfer vectors from one VectorSchemaRoot to another one?  Or is one VectorSchemaRoot intented to be shared across multiple processors?

Reply:
We could get a new VectorSchemaRoot instance by VectorShemaRoot.slice API(do transfer for its vectors), but I am not sure if this is the right usage considering VectorSchemaRoot != recordBatch